src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java

```diff
@@ -1017,6 +1027,18 @@ public void saveData()
 		throw new UnimplementedOperationException();
 	}
 
+	/**
+	 * This function simulates data loss from consequences of
+	 * qutting the server by player. It should be used only in ServerMock.
+	 */
+	public void simulateDataLossAfterSave()
+	{
+		hiddenPlayers.clear();
+		hiddenPlayersDeprecated.clear();
+		heardSounds.clear();
+		sneaking = sprinting = false;
+	}
```

Comment:
hmmmmmmmm, not too sure what to think of this method here

Reply:
Because this removePlayer function doesn't emulate saving player data in any state (all the data stays in PlayerMock object), instead of saving and reading again needed variables, it just deletes the ones that aren't saved by a server after player quits, so it's faster and it takes much less code